plugins/qcheck-stm/src/stm_of_ir.ml

```diff
@@ -676,29 +728,36 @@ let pp_cmd_case value =
   in
   let* rhs =
     let name = str_of_ident value.id in
-    let* pp_args =
-      concat_map
-        (fun (ty, id) ->
+    let rec aux ty args =
+      match (ty.ptyp_desc, args) with
+      | Ptyp_arrow (_, l, r), xs when Cfg.is_sut config l ->
+          let* fmt, pps = aux r xs in
+          ok ("%s" :: fmt, estring "sut" :: pps)
```

Comment:
If `fmt` directly contained `sut` instead of `%s`, we could keep `pp_args` unchanged.
On the other hand, maybe your proposition is an altogether better approach, to build `fmt` and `pp_args` at the same time.

Reply:
We can indeed spare the `%s` and `estring "sut"`